azure/compute/src/main/java/org/jclouds/azurecompute/compute/suppliers/LocationSupplier.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This file has been modified by Microsoft Open Technologies, Inc.
+ * Modifications copyright (c) Microsoft Open Technologies, Inc.
+ */
+
+package org.jclouds.azurecompute.compute.suppliers;
+
+import com.google.common.base.Function;
+import com.google.common.base.Supplier;
+import com.google.common.collect.FluentIterable;
+import com.google.inject.Provides;
+import com.microsoft.windowsazure.management.models.LocationsListResponse;
+import com.microsoft.windowsazure.management.models.RoleSizeListResponse;
+import org.jclouds.azurecompute.domain.VirtualMachine;
+import org.jclouds.azurecompute.domain.OSImage;
+import org.jclouds.azurecompute.domain.VirtualMachine;
+import org.jclouds.collect.Memoized;
+import org.jclouds.compute.ComputeServiceAdapter;
+import org.jclouds.compute.domain.Hardware;
+import org.jclouds.domain.Location;
+import org.jclouds.location.Provider;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.google.common.base.Predicates.notNull;
+
+@Singleton
+@Memoized
+public class LocationSupplier implements Supplier<Set<? extends Location>> {
+    private ComputeServiceAdapter<VirtualMachine, RoleSizeListResponse.RoleSize, OSImage, LocationsListResponse.Location> adapter;
+    private Function<LocationsListResponse.Location, Location> transformer;
+
+    @Inject
+    public LocationSupplier(ComputeServiceAdapter<VirtualMachine, RoleSizeListResponse.RoleSize, OSImage,
+            LocationsListResponse.Location> adapter, Function<LocationsListResponse.Location, Location> transformer) {
+        this.adapter = adapter;
+        this.transformer = transformer;
+    }
+
+    @Override
+    public Set<? extends Location> get() {
+        return FluentIterable.from(adapter.listLocations()).filter(notNull()).transform(transformer).toSet();
+//            return transformGuardingNull(adapter.listLocations(), transformer);
+    }
+}
```

Comment:
There is no need to provide a custom location supplier, as the adapter already has a method that lists them. Just add this binding to the `AzoreComputeServiceContextModule` configure method:
```java
install(new LocationsFromComputeServiceAdapterModule<VirtualMachine, RoleSizeListResponse.RoleSize, OSImage, LocationsListResponse.Location>() {
});
```
And then you can remove this class. jclouds will extract the locations by calling the compute service implementation and the corresponding transformation function.